phac-nml/irida-next
# update: Confirm form confirmation

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Updates the confirm component to no longer accept a form.  Instead the component handles the form itself for validation against a given value.

Example (see LookBook example below):
```html
<%= button_to "Delete project",
"#",
data: {
  turbo_confirm: "",
  turbo_content: ".custom-content",
  confirm_value: "Project X"
},
class: "button button--state-destructive button--size-default" %>
```

Here we provide a custom content telling the user what they want (within the .custom-content div), and the value to validate against in `confirm_value`

![image](https://github.com/phac-nml/irida-next/assets/11295750/b3d58cbd-9d71-4907-b863-41ce4c1893de)

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/11295750/f81c435e-de24-40b1-96bd-562f23372a4d)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

[See LookBook confirmation > with confirm value](http://localhost:3000/rails/lookbook/inspect/confirmation/with_confirm_value?_display=%257B%2522theme%2522%253A%2522light%2522%257D)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

  the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.
